test/Misc/verify-swift-feature-testing.test-sh

```diff
@@ -77,133 +61,105 @@ def find_test_files_with_marker_usage(swift_src_root):
     return output.splitlines()
 
 
-def find_run_lines(test_file):
+def find_run_and_requires_lines(test_file):
+    # Be careful to not use RUN or REQUIRES with a colon after them or Lit
+    # will pick them up.
     output = subprocess.check_output(
         [
             "grep",
             "--extended-regexp",
             "--no-filename",
             "-e",
             "RUN[:]",
-            str(test_file),
-        ],
-        text=True,
-    )
-    return output.splitlines()
-
-
-def find_requires_lines(test_file):
-    output = subprocess.check_output(
-        [
-            "grep",
-            "--extended-regexp",
-            "--no-filename",
             "-e",
             "REQUIRES[:]",
-            str(test_file),
+            test_file,
         ],
         text=True,
     )
     return output.splitlines()
 
 
-def check_existing_requires(test_file, feature):
+def check_existing_error_message_checks(test_file, feature):
     returncode = subprocess.call(
         [
             "grep",
             "--extended-regexp",
             "--quiet",
             "-e",
-            "REQUIRES[:].*swift_feature_" + feature,
-            str(test_file),
+            "requires '-enable-(experimental|upcoming)-feature " + feature + "'",
+            test_file,
         ]
     )
     return returncode != 0
 
 
-def check_existing_feature_usage(test_file, feature):
-    returncode = subprocess.call(
-        [
-            "grep",
-            "--extended-regexp",
-            "--quiet",
-            "-e",
-            (
-                "RUN[:].*-enable-(experimental|upcoming)-feature (-Xfrontend )?"
-                + re.escape(feature)
-            ),
-            str(test_file),
-        ]
-    )
-    return returncode != 0
+def check_test_file(test_file, existing_swift_features):
+    run_features = set()
+    requires_features = set()
 
+    for line in find_run_and_requires_lines(test_file):
+        run_features.update(feature for feature in ENABLE_FEATURE_RE.findall(line))
+        requires_features.update(
+            feature for feature in FEATURE_LIT_MARKER_RE.findall(line)
+        )
 
-def check_existing_error_message_checks(test_file, feature):
-    returncode = subprocess.call(
-        [
-            "grep",
-            "--extended-regexp",
-            "--quiet",
-            "-e",
-            "requires '-enable-(experimental|upcoming)-feature " + feature + "'",
-            str(test_file),
-        ]
-    )
-    return returncode != 0
+    had_error = False
 
+    # First check for unknown features.
 
-def check_test_file_feature_usage(test_file, existing_swift_features):
-    run_lines = find_run_lines(test_file)
-    features = set(
-        feature for line in run_lines for feature in FEATURE_USAGE_RE.findall(line)
-    )
-    num_failures = 0
-    for feature in features:
-        # First, check this is a valid feature
+    for feature in run_features.copy():
```

Comment:
Since `run_features` and `existing_swift_features` are both `set`, this can probably done with some set operation (`run_features - existing_swift_features`?).

Same below with `requires_feature`.